### PR TITLE
feat(recover identity): Connect ion-identity-client to the app

### PR DESCRIPTION
**Pull request by ice-tychon**
### What does this PR do?
This PR connects UI to the Ion Identity Client to let a user create, validate and use recovery key.

### Changes brought by this PR
- Added a new library `extended_text` to use "ellipsis middle" (so instead of `very long ...` it can be `very ... string`)
- Renamed a few things (enum `RecoveryKeys -> RecoveryKeyProperty`, `RecoveryKeysSavePage -> CreateRecoveryKeyPage`, `RecoveryKeysInputPage -> ValidateRecoveryKeyPage`)
- Updated `AuthProvider` a lot:
  * rename fields in `AuthState`
  * remove `rehydrate` 'cause now the authorized users are stored in the `ion-identity-client` lib
  * extracted `signIn` and `signOut` to separate providers `LoginActionNotifier` and `RegisterActionNotifier`; updated other code accordingly to use new providers
  * added a few additional providers to simplify the code: `SelectedIdentityKeyNameStore` and `authenticatedIdentityKeyNamesStream`
- Updated `CreateRecoveryKeyPage` and `ValidateRecoveryKeyPage` to use Ion Identity Client

### CODE REVIEW

**Review by ice-kreios — DISMISSED** [review]

File: lib/app/features/auth/views/pages/restore_recovery_keys/restore_recovery_keys_page.dart [hunk 1]
```diff
@@ -1,16 +1,71 @@
...
-  Widget build(BuildContext context) {
-    return RecoveryKeysInputContainer(
-      onContinuePressed: () => TwoFaOptionsRoute().push<void>(context),
+  Widget build(BuildContext context, WidgetRef ref) {
+    final ionClient = ref.watch(ionApiClientProvider.future);
+
+    final recoverUserResult = useState<RecoverUserResult?>(null);
+
+    if (recoverUserResult.value == null) {
+      return FutureBuilder<IonApiClient>(
```
Comment (ice-alcides) [inline lib/app/features/auth/views/pages/restore_recovery_keys/restore_recovery_keys_page.dart:26]: Is this `FutureBuilder` temporary solution?

File: lib/app/features/auth/providers/auth_provider.dart [hunk 2]
```diff
@@ -1,111 +1,99 @@
...
-      localStorage.remove(_currentUserIdKey),
-    ]);
+  yield* ionClient.authorizedUsers;
+}
 
-    state = const AsyncData(AuthState(authenticatedUserIds: [], currentUserId: null));
-  }
+@Riverpod(keepAlive: true)
+class SelectedIdentityKeyNameStore extends _$SelectedIdentityKeyNameStore {
+  static const String _currentUserIdKey = 'Auth:currentUserId';
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply lib/app/features/auth/providers/auth_provider.dart]: Done. Update the name and value.

File: lib/app/features/protect_account/backup/data/providers/recovery_key_data_provider.dart
[the remark this answers is not in the document]
Comment (ice-tychon) [reply lib/app/features/protect_account/backup/data/providers/recovery_key_data_provider.dart:1]: Done. Moved to another folder.

File: lib/app/features/auth/views/pages/restore_recovery_keys/restore_recovery_keys_page.dart (hunk 1 above)
Comment (ice-tychon) [reply lib/app/features/auth/views/pages/restore_recovery_keys/restore_recovery_keys_page.dart]: Done. Update the code to remove `FutureBuilder`.

File: lib/app/features/auth/providers/auth_provider.dart [hunk 3]
```diff
@@ -1,111 +1,99 @@
...
-  Future<void> signOut() async {
-    state = const AsyncValue.loading();
+  void selectUser(String identityKeyName) {
+    ref.read(selectedIdentityKeyNameStoreProvider.notifier).selectIdentityKeyName(identityKeyName);
+  }
+}
 
-    await Future<void>.delayed(const Duration(seconds: 1));
+@riverpod
+String currentUserIdSelector(CurrentUserIdSelectorRef ref) {
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply lib/app/features/auth/providers/auth_provider.dart:67]: We use `currentUserIdSelectorProvider` in 14 files. Should I update the name in the current PR or create a separate one for this?

**Review by ice-kreios — APPROVED** [review]

**Review by ice-alcides — APPROVED** [review]

**Pull request merged**